htpy/html2htpy.py

```diff
@@ -160,28 +162,40 @@ def serialize(self, *, shorthand_id_class: bool, use_h_prefix: bool) -> str:
 
 
 class Formatter(ABC):
+    error_return_code: int
+
     @abstractmethod
     def format(self, s: str) -> str:
         raise NotImplementedError()
 
 
 class BlackFormatter(Formatter):
+    error_return_code = 123
+
     def format(self, s: str) -> str:
         result = subprocess.run(
             ["black", "-q", "-"],
             input=s.encode("utf8"),
             stdout=subprocess.PIPE,
         )
+        if result.returncode == self.error_return_code:
+            _printerr("Black failed to parse the input. The output will be left unformatted.")
+            return s
         return result.stdout.decode("utf8")
 
 
 class RuffFormatter(Formatter):
+    error_return_code = 2
+
     def format(self, s: str) -> str:
         result = subprocess.run(
             ["ruff", "format", "-"],
             input=s.encode("utf8"),
             stdout=subprocess.PIPE,
         )
+        if result.returncode == self.error_return_code:
+            _printerr("Ruff failed to parse the input. The output will be left unformatted.")
+            return s
```

Comment:
question: when does this happen? is this not always a bug in html2htpy? if it produces invalid python code? 😬 

suggestion: can we clarify this and add something like this to the error message: "This is likely a bug in html2htpy. Please report this as an issue to htpy: https://github.com/pelme/htpy/issues".